factory/api/apiResolverFactory_test.go

```diff
@@ -452,3 +497,51 @@ func TestCreateApiResolver_createScQueryElement(t *testing.T) {
 	})
 
 }
+
+func TestCreateApiResolver_createArgsSCQueryService(t *testing.T) {
+	t.Parallel()
+
+	t.Run("sovereign chain should add systemVM", func(t *testing.T) {
+		t.Parallel()
+
+		args := createMockSCQueryElementArgs()
+		args.ChainRunType = common.ChainRunTypeSovereign
+
+		argsScQueryService, err := api.CreateArgsSCQueryService(args)
+		require.Nil(t, err)
+		require.NotNil(t, argsScQueryService.VmContainer)
+
+		require.Equal(t, 2, argsScQueryService.VmContainer.Len())
+
+		svm, err := argsScQueryService.VmContainer.Get(vmFactory.SystemVirtualMachine)
+		require.Nil(t, err)
+		require.NotNil(t, svm)
+		require.Equal(t, "*process.systemVM", fmt.Sprintf("%T", svm))
+
+		wasmvm, err := argsScQueryService.VmContainer.Get(vmFactory.WasmVirtualMachine)
+		require.Nil(t, err)
+		require.NotNil(t, wasmvm)
+		require.Equal(t, "*hostCore.vmHost", fmt.Sprintf("%T", wasmvm))
+	})
+	t.Run("regular chain should NOT add systemVM", func(t *testing.T) {
```

Comment:
This is a good test for regular shard chain. We should also check the factory for a metachain shard chain in an unit test, if it doesn't already exist